lib/index.ts

```diff
@@ -571,4 +574,40 @@ export default class WalletClient {
       this.eventListeners.get(event)!.forEach((listener) => listener(data));
     }
   }
+
+  private populateWindowObject() {
+    if (typeof window !== 'undefined') {
+      const createMethodWrapper = (method: Function) => {
+        return async (...args: any[]) => {
```

Comment:
Avoid type any, because user of this lib will not have understanding of what args have to be passed to each of the wrapped functions. Try to infer type from generic either passed alongside with wrapper call or dynamically get the type of function's arguments.

Reply:
otherwise people will need to go into source code and find those types themselves, which is obviously not the flow that they expect to see